### PR TITLE
feat: custom bad gateway statuses

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

File: src/chain.ts [hunk 1]
```diff
@@ -118,7 +119,10 @@ export const chain = (
             if (isPlain) {
                 sourceSocket.end();
             } else {
-                sourceSocket.end(createHttpResponse(502, ''));
+                const unauthorized = [401, /* 403, */ 407];
+                const status = unauthorized.includes(response.statusCode!) ? 597 : 590;
```
Comment (jancurn) [inline src/chain.ts:123]: It would be better and less error-prone to put these numbers in a constant

File: src/statuses.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export const statuses: {[code: string]: number | undefined} = {
```
Comment (jancurn) [inline src/statuses.ts:1]: Comment?

**Review by jancurn — APPROVED** [review]
Neat idea, just a few comments

File: src/chain.ts (hunk 1 above)
Comment (szmarczak) [reply src/chain.ts:123]: Doesn't matter but I guess it's more readable, will do that 👍

Comment (szmarczak) [reply src/chain.ts:123]: Doesn't matter but I guess it's more readable, will do that 👍

File: src/statuses.ts (hunk 2 above)
Comment (szmarczak) [reply src/statuses.ts:1]: What needs to be commented here?

Comment (jancurn) [reply src/statuses.ts:1]: just what is this lookup table for

File: src/chain.ts (hunk 1 above)
Comment (jancurn) [reply src/chain.ts:123]: I mean to have all the new codes with a special meaning in a constant dictionary, just to make it clear what they mean, without having to look into README

File: src/statuses.ts (hunk 2 above)
Comment (szmarczak) [reply src/statuses.ts:1]: I guess I can just rename `code` to `errorCode` and alias `number` as `HttpStatusCode`?

**Review by jirimoravcik — APPROVED** [review]

**Comment by szmarczak** [comment]
@jancurn Added comments for https://github.com/apify/proxy-chain/blob/custom-bad-gateway/src/statuses.ts PTAL :)

**Review by jancurn — APPROVED** [review]

**Comment by jancurn** [comment]
Great, thanks @szmarczak

**Pull request merged**